NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Ticket #10 - Create a Category

# Description
This PR is for [Ticket #10](https://trello.com/c/Yxs0p6sM/10-10-create-a-category), which should allow users to add to the list of available categories. 

> As an admin I would like to be able to create a new category so that I can give authors the ability to better classify their posts.
> 
> **Given** an admin is on the Category list page
> **When** they select the `Create Category` button
> **Then** they should be directed to a form in which they can enter a new category name
> 
> **Given** an admin has entered a Category name
> **When** they click the `Save` button
> **Then** a new category should be saved to the database
> **And** the admin should be directed to the Category list page
> 
> > **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.



# Testing Instructions
1. Commit or stash whatever you're currently working on
2. Git fetch --all
3. Git checkout ac-createNewCategory
4. If you don't see the most recent version of my files, git pull origin ac-createNewCategory
5.  If you need to create the Tabloid database, [use this file to do so](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
6. If you don't have the Tabloid seed data, [use this file to get it
](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
7. Execute Tabloid from inside of Visual Studio
![Tabloid](https://user-images.githubusercontent.com/106983427/202474349-d57baf9b-323c-4406-b75a-d8f99a886603.png)
8.  Run NPM Start from inside of the Tabloid client folder
![client](https://user-images.githubusercontent.com/106983427/202473962-42d0b50f-1420-4b34-a522-464dc9fc11d5.png)
9. Login using an email that you know is there and works for you (example: foo@bar.com from seed data)
10. Once logged-in, use the navbar to navigate to category management ("/category")
11. Click the create category button located near the top of the page 
12. You should be redirected to a page ("/category/add") containing a single input field and a save button 
13. Enter a new category name and click save to be redirected to the list of categories
14. You should see that your newly-created category has been added to the list

# Checklist:
